src/main/java/core/CoreConstants.java

```diff
@@ -32,7 +32,7 @@
     public final static int backgroundImgHash = Hash.GetInstance().hash("backgroundImg");
 
     public enum GameEvents {
-        GAME_OVER, ROUND_OVER, TURN_OVER, ACTION_CHOSEN
+        ABOUT_TO_START, GAME_OVER, ROUND_OVER, TURN_OVER, ACTION_CHOSEN, GAME_SEQUENCE_OVER
```

Comment:
Is GAME_SEQUENCE_OVER meant to capture action sequences, or several games being played? I'm not sure that's a clear name for it.

Reply:
Yes - just checked that TTT GUI still works.

I was also a little uncomfortable with GAME_SEQUENCE_OVER. I've amended the implementation to exclude this, so that GameEvents are actually all genuinely Game-Events. The end of a sequence of games is now controlled by a new (optional) method on the IGameListener interface.